index.js

```diff
@@ -277,11 +284,17 @@ function onReady () {
             elements[json.targetID].unmaximize()
             break;
             case consts.eventNames.windowCmdUpdateCustomOptions:
-            windowOptions[json.targetID] = json.windowOptions
-            client.write(json.targetID, consts.eventNames.windowEventUpdatedCustomOptions, json.windowOptions)
+            client.write(json.targetID, consts.eventNames.windowEventUpdatedCustomOptions, json.windowOptions);
             break;
             case consts.eventNames.windowCmdWebContentsExecuteJavascript:
-            elements[json.targetID].webContents.executeJavaScript(json.code).then(() => client.write(json.targetID, consts.eventNames.windowEventWebContentsExecutedJavaScript));
+            elements[json.targetID].webContents.executeJavaScript(json.code)
+                .then(() => client.write(json.targetID, consts.eventNames.windowEventWebContentsExecutedJavaScript));
+            break;
+            case consts.eventNames.windowCmdWebContentsSetProxy:
+            elements[json.targetID].webContents.session.clearAuthCache().then(() => {
+                elements[json.targetID].webContents.session.setProxy({proxyRules: json.proxy})
```

Comment:
I'd rather the json to provide the complete options given to `setProxy`: `elements[json.targetID].webContents.session.setProxy(json.proxy)`